include/unifex/sender_for.hpp

```diff
@@ -0,0 +1,116 @@
+ * limitations under the License.
+ */
+#pragma once
+
+#include <unifex/config.hpp>
+#include <unifex/sender_concepts.hpp>
+#include <unifex/tag_invoke.hpp>
+#include <unifex/type_traits.hpp>
+
+#include <unifex/detail/prologue.hpp>
+
+namespace unifex
+{
+  namespace _sf
+  {
+    template <typename Query>
+    struct _property {
+      const typename Query::value_type& operator()(const typename Query::key_type &) const noexcept {
+        return query_.value;
+      }
+      UNIFEX_NO_UNIQUE_ADDRESS Query query_;
+    };
+    template <typename... Queries>
+    struct _ctx {
+      struct type : _property<Queries>... {
+        using _property<Queries>::operator()...;
+      };
+    };
+    template <typename... Queries>
+    using _context = typename _ctx<Queries...>::type;
+
+    template <const auto& C, typename Sender, typename Context>
+    struct sender_for {
+      explicit sender_for(Sender snd, Context ctx)
+          noexcept(
+              std::is_nothrow_move_constructible_v<Sender> &&
+              std::is_nothrow_move_constructible_v<Context>)
+        : snd_((Sender&&) snd), ctx_{(Context&&) ctx} {}
+
+      // Forward all tag_invokes:
+      template (typename CPO, typename Self, typename... Args)
+        (requires same_as<sender_for, remove_cvref_t<Self>> AND
+            (!callable<const Context&, CPO>) AND
+            callable<CPO, member_t<Self, Sender>, Args...>)
+      UNIFEX_ALWAYS_INLINE
+      friend decltype(auto) tag_invoke(CPO cpo, Self&& self, Args&&... args)
+          noexcept(is_nothrow_callable_v<CPO, member_t<Self, Sender>, Args...>) {
+        return ((CPO&&) cpo)(((Self&&) self).snd_, (Args&&) args...);
+      }
+
+      // Handle custom property queries by consulting the context:
+      template (typename CPO)
+        (requires callable<const Context&, CPO>)
+      UNIFEX_ALWAYS_INLINE
+      friend decltype(auto) tag_invoke(CPO cpo, const sender_for& self) noexcept {
+        return self.ctx_(cpo);
+      }
+
+      const Sender& base() const & noexcept {
+        return snd_;
+      }
+
+      Sender&& base() && noexcept {
+        return (Sender&&) snd_;
+      }
+     private:
+      Sender snd_;
+      UNIFEX_NO_UNIQUE_ADDRESS Context ctx_;
+    };
+
+    template <auto& CPO>
+    struct _make_sender_for {
+      template (typename Sender, typename... Queries)
+        (requires sender<Sender>)
+      sender_for<CPO, remove_cvref_t<Sender>, _context<remove_cvref_t<Queries>...>>
+      operator()(Sender&& snd, Queries&&... queries) const
+          noexcept(
+              std::is_nothrow_constructible_v<remove_cvref_t<Sender>, Sender> &&
+              (std::is_nothrow_constructible_v<remove_cvref_t<Queries>, Queries> &&...)) {
+        return sender_for<CPO, remove_cvref_t<Sender>, _context<remove_cvref_t<Queries>...>>{
+            (Sender&&) snd, _context<remove_cvref_t<Queries>...>{{(Queries&&) queries}...}};
+      }
+    };
+  } // namespace _sf
+
+  using _sf::sender_for;
```

Comment:
This exists only in support of _Scheduler_ affinity; it might make sense to defer porting this file until you port that feature.

Reply:
Removed.